sql/sql_parse.cc

```diff
@@ -7876,9 +7876,17 @@ static uint kill_one_thread(THD *thd, my_thread_id id, bool only_kill_query)
                             tmp->m_security_ctx->ip().str);
 
 #ifdef WITH_WSREP
-    if ((((thd->security_context()->check_access(SUPER_ACL)) && !is_utility_connection) ||
-         thd->security_context()->user_matches(tmp->security_context())) &&
-        !wsrep_thd_is_BF((void *)tmp, true))
+    if (wsrep_thd_is_in_to_isolation(tmp)) {
+      /* There is nothing special about the use of ER_QUERY_INTERRUPTED,
+         I just needed a different error code from ER_KILL_DENIED_ERROR.
+
+         See sql_kill() to see the other half of the error processing.
+      */
+      error= ER_QUERY_INTERRUPTED;
+    }
+    else if (((thd->security_context()->check_access(SUPER_ACL)) ||
```

Comment:
Looks like `&& !is_utility_connection)` is omitted. It is causing build failures
```
[ 87%] Building CXX object sql/CMakeFiles/sql.dir/mysqld.cc.o
/home/venki/work/pxc/57/sql/sql_parse.cc: In function ‘uint kill_one_thread(THD*, my_thread_id, bool)’:
/home/venki/work/pxc/57/sql/sql_parse.cc:7873:16: error: unused variable ‘is_utility_connection’ [-Werror=unused-variable]
     const bool is_utility_connection =
                ^~~~~~~~~~~~~~~~~~~~~
[ 87%] Building CXX object sql/CMakeFiles/sql.dir/mysqld_thd_manager.cc.o
```

Also, looks like this branch is not based on GCA. Please check it once.